framework/src/meshgenerators/ConvertNodeSetSideSetGenerator.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ConvertNodeSetSideSetGenerator.h"
+#include "MooseMeshUtils.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/mesh_refinement.h"
+#include "CastUniquePointer.h"
+
+registerMooseObject("MooseApp", ConvertNodeSetSideSetGenerator);
+
+InputParameters
+ConvertNodeSetSideSetGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addClassDescription(
+      "Mesh generator which constructs side sets from node sets, and vise versa");
+  params.addRequiredParam<MeshGeneratorName>("input",
+                                             "Input mesh the operation will be applied to");
+  params.addRequiredParam<bool>("convert_node_list_from_side_list",
+                                "Flag signaling if nodes should be constructed from sides");
+  params.addRequiredParam<bool>("convert_side_list_from_node_list",
+                                "Flag signaling if sides should be constructed from nodes");
```

Comment:
I am not a fan of using booleans to combine capabilities. The MOOSE-way of doing it is to have two separate classes. A `NodeSetToSideSetGenerator` and a `SideSetToNodeSetGenerator`. They could be deriving from the same parent class to reduce code duplication.

Reply:
On one hand selecting the nodesets / sidesets would be nice and make it more targeted.

On the other hand the other mesh generators that @RocksSavage are more important for our applications so I dont want to spend too long on these

So I'd say address the other comments and we'll advise later when we have good progress on the others